platform/machine/oci/cluster.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+
+	"github.com/coreos/pkg/capnslog"
+
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/platform/api/oci"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+var (
+	plog = capnslog.NewPackageLogger("github.com/coreos/mantle", "platform/machine/oci")
+)
+
+type cluster struct {
+	*platform.BaseCluster
+	api *oci.API
+}
+
+func NewCluster(opts *oci.Options, rconf *platform.RuntimeConfig) (platform.Cluster, error) {
+	api, err := oci.New(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	bc, err := platform.NewBaseCluster(opts.BaseName, rconf, "")
+	if err != nil {
+		return nil, err
+	}
+
+	oc := &cluster{
+		BaseCluster: bc,
+		api:         api,
+	}
+
+	return oc, nil
+}
+
+func (oc *cluster) vmname() string {
+	b := make([]byte, 5)
+	rand.Read(b)
+	return fmt.Sprintf("%s-%x", oc.Name(), b)
+}
+
+func (oc *cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error) {
```

Comment:
To avoid confusion, this should probably fail if `userdata` is not Ignition-compatible.  `userdata.IsIgnition()` is almost the right test, except it returns `false` for CLCs, which is arguably a bug.  (Its only other caller should probably handle CLCs too.)  Maybe fix it and rename to `IsIgnitionCompatible()`?

Reply:
The way other platforms seem to be doing ignition detection is to call the `IsIgnition` function on the `conf.Conf` object after the `conf.UserData` has been rendered. Do we want to change the underlying `IsIgnition` function of `conf.UserData` and change the way those platforms are doing their detection or mimic that pattern?

Reply:
As discussed OOB, we can do the detection either on the `UserData` or the `Conf`, but `UserData.IsIgnition()` should be fixed regardless.